File: src/main/java/org/intense/ast/IfConditionNode.java
package org.intense.ast;

import org.intense.Env;
import org.intense.Types.BoolVal;
import org.intense.Types.NullVal;
import org.intense.Types.Value;

public class IfConditionNode extends ASTNode{
    private ASTNode ifExp;
    private ASTNode ifBody;
    private ASTNode elseBody;

    public IfConditionNode(ASTNode ifExpr, ASTNode dtifBody, ASTNode dtelseBody) {
        ifExp = ifExpr;
        ifBody = dtifBody;
        elseBody = dtelseBody;
    }


    @Override
    public Value eval(Env env) {
        Value exp = ifExp.eval(env);
        if (exp instanceof BoolVal boolVal)
        {
            if(boolVal.value)
            {
                if (ifBody != null) {
                 return ifBody.eval(env);
                }
            }
            else {
                if (elseBody != null) {
                return elseBody.eval(env);
                }
            }
            return new NullVal();
        }
        else {
            throw new RuntimeException("if expression must return a boolean value as output");
        }
    }
}

File: src/main/java/org/intense/ast/IfConditionNode.kt
package org.intense.ast;

import org.intense.Env
import org.intense.Types.NullVal
import org.intense.Types.Value

class IfConditionNode(ifExpr: ASTNode, dtifBody: ASTNode, dtelseBody: ASTNode) : ASTNode() {
    private var ifExp: ASTNode = ifExpr
    private var ifBody: ASTNode? = dtifBody
    private var elseBody: ASTNode? = dtelseBody

    override fun eval(env: Env): Value {
        val expr: StringBuilder = StringBuilder()
        expr.append("\nif (" + ifExp.eval(env) + "){")
        if (ifBody != null) {
            expr.append(ifBody?.eval(env))
            expr.append("}")
            if (elseBody != null) {
                expr.append("else {")
                elseBody?.eval(env)
                expr.append("}")
            }
        }
        return NullVal();
    }
}
